### PR TITLE
Migrate AnalysisCategories to DX

### PR DESCRIPTION
**Pull request by Bugerman58**
## Description of the issue/feature this PR addresses

This PR migrates the AT AnalysisCategory and AnalysisCategories to Dexterity types for Python 3 compatibility.

## Current behavior before PR

AnalysisCategory and AnalysisCategories are content types based on the Archetypes framework

## Desired behavior after PR is merged

AnalysisCategory and AnalysisCategories are content types based on the Dexterity framework

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by Bugerman58** [comment]
@xispa Hi, this PR ready for review. pls, check it

File: src/bika/lims/workflow/analysiscategory/guards.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import api
+from bika.lims.catalog.bikasetup_catalog import SETUP_CATALOG
```
Comment (xispa) [inline src/bika/lims/workflow/analysiscategory/guards.py]: better import the catalog id from `senaite.core` instead:

```python
from senaite.core.catalog import SETUP_CATALOG
```

File: src/bika/lims/workflow/analysiscategory/guards.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def guard_deactivate(analysis_category):
```
Comment (xispa) [inline src/bika/lims/workflow/analysiscategory/guards.py:25]: I see you replaced that old-school `workflow_script_deactivate` by this guard handler, good job!

File: src/senaite/core/content/analysiscategory.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        limit=5,
+    )
+    department = UIDReferenceField(
+        title=_(
+            "title_analysiscategory_department",
+            default="Department",
+        ),
+        description=_(
+            "description_analysiscategory_department",
+            default="Select the responsible department",
```
Comment (xispa) [inline src/senaite/core/content/analysiscategory.py]: Please do `unicode` for `i18n` resources. See this discussion: https://github.com/senaite/senaite.core/pull/2566#pullrequestreview-2089339759

File: src/senaite/core/content/analysiscategory.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    )
+
+    # Department reference
+    directives.widget(
+        "department",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query="get_department_query",
+        columns=get_default_columns,
+        limit=5,
```
Comment (xispa) [inline src/senaite/core/content/analysiscategory.py]: Remove `limit=5`, it is the default. See https://github.com/senaite/senaite.core/blob/2.x/src/senaite/core/z3cform/widgets/queryselect/widget.py#L118

**Review by xispa — CHANGES_REQUESTED** [review]
Thanks @Bugerman58 !

There is a [subscriber that reindexes all sample analyses when the title of the category changes](https://github.com/senaite/senaite.core/blob/2.x/src/bika/lims/subscribers/objectmodified.py#L50). This subscriber won't be called because the event is only triggered for AT content types. I do know that the dance is weird, takes ages to complete and is something we need to fix. But keep it for now, please. I will open a PR after yours to remove the `getCategoryTitle` metadata and fix that thing. Meanwhile, please create a new subscriber specific for analysiscategory at `senaite.core.subscribers`.

**Review by xispa — APPROVED** [review]
Excellent, thanks!

**Pull request merged**